### PR TITLE
CLN: Consolidate numba facilities

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Precursor to https://github.com/pandas-dev/pandas/issues/31845 and other numba engine additions, creates `pandas/core/numba_.py` (open to move elsewhere) as a shared place for common numba operations like default parameters and jitting functions.

### CODE REVIEW

**Comment by WillAyd** [comment]
Maybe better placed in pandas/core/util? @jreback

**Comment by mroeschke** [comment]
Good point. I can move it to util.

File: pandas/core/util/numba_.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""Common utilities for Numba operations"""
```
Comment (jbrockmendel) [inline pandas/core/util/numba_.py]: no strong opinion, but should this go in pandas.compat?

**Comment by jreback** [comment]
seems good enough place for now. we can see how / if this grows and move around later. e.g. this is conceptually like pandas/core/computation stuff (though not nearly as big).

**Pull request merged**